### DISCUSSION TITLE
Hyperparameter Optimization for predict_tile

### DISCUSSION THREAD

**Discussion by mtaniguchiking** [weecology/DeepForest#472] [General]
### Hello! I'm evaluating DeepForest's [predict_tile](https://deepforest.readthedocs.io/en/latest/source/deepforest.html#deepforest.main.deepforest.predict_tile) as a CV tree detection method for [OFO](https://openforestobservatory.org/) and would appreciate insight/feedback on my current approach:

As a starting point for the evaluation, I'm optimizing `predict_tile`'s hyperparameters on a forest orthomosaic to gauge the effectiveness of `predict_tile` and to get a sense for how the performance (i.e. f-score) of predict_tile changes depending on its hyperparameters. Before considering multiple parameters, I began by exploring the influence of individual parameters - as an example, here is a preliminary result from a range of patch_size values:
<img width="500" alt="image" src="https://github.com/weecology/DeepForest/assets/112134530/58b73d8c-f1ff-4f81-97e5-9da810290341">

My current approach for hyperparameter fine-tuning involves running `predict_tile` on an [orthomosaic](https://ucdavis.app.box.com/s/4uqts0zc8h52znl5avurjwntm2bn4w92/file/1236320290239) with different hyperparameter combinations before comparing the DeepForest tree detections against a ground-measured reference tree map.

My plan involves performing the following searches for hyperparameter fine-tuning:
Random Search with NMS (`use_soft_nms = False`):
- Parameters to optimize: patch_size, patch_overlap, and iou_threshold, orthomosaic resolution (image rescale factor)

I’m curious about the role of `thresh` when `use_soft_nms = False`. Documentation indicates `thresh` only filters bboxes after soft NMS is performed (therefore, it is not relevant when `use_soft_nms = False`), but I would like to confirm this. I’m also more generally interested in other considerations (specific to DeepForest) I should take into account as I optimize these parameters.

**Comment by bw4sz** [comment]
Closing, since was responded by email. #471 opened issue.
